compile/api/src/main/scala/sbt/ClassToAPI.scala

```diff
@@ -127,14 +133,54 @@ object ClassToAPI {
   def upperBounds(ts: Array[Type]): api.Type =
     new api.Structure(lzy(types(ts)), lzyEmptyDefArray, lzyEmptyDefArray)
 
-  def fieldToDef(enclPkg: Option[String])(f: Field): api.FieldLike =
+  /** Parses the constant value represented by the given ConstantValue AttributeInfo. */
+  private def constantPoolConstantValue(cf: ClassFile, ai: classfile.AttributeInfo): AnyRef = {
+    assert(ai.name.exists(_ == "ConstantValue"), s"Non-ConstantValue attribute not supported: ${ai}")
+    import classfile.Constants._
+    cf.constantPool(classfile.Parser.entryIndex(ai)) match {
+      case classfile.Constant(ConstantString, nextOffset, _, _) =>
+        // follow the indirection from ConstantString to ConstantUTF8
+        val nextConstant = cf.constantPool(nextOffset)
+        nextConstant.value.getOrElse {
+          throw new RuntimeException(s"Empty UTF8 value in constant pool: $nextConstant")
+        }
+      case constant @ classfile.Constant((ConstantFloat | ConstantLong | ConstantDouble | ConstantInteger), _, _, ref) =>
+        ref.getOrElse {
+          throw new RuntimeException(s"Empty primitive value in constant pool: $constant")
```

Comment:
Again, ideally the error message should offer guidance for how to fix.  (the fallback in my mind is just assume corrupted class file and ask to clean).  Something like:

`s"Empty primitive value in constant pool: $constant \n  This may mean your classfiles are corrupted.  Please clean and try again"`